### PR TITLE
feat: Disable rollup and transform for pluggable dataformat indices

### PR DESCRIPTION
**Pull request by KhishorekumarBS**
### Description
Block creation of rollup and transform jobs when the source index has index.pluggable.dataformat.enabled = true. Indices with pluggable dataformat use a non-Lucene storage format that is currently incompatible with rollup and transform operations.

Validation is added in three places:
- TransportIndexRollupAction (standalone rollup creation)
- TransportIndexTransformAction (standalone transform creation)
- AttemptCreateRollupJobStep (ISM policy rollup action)

A shared utility PluggableDataFormatUtils is introduced to centralize the setting check logic with unit tests covering all edge cases.

### Check List
- [✅ ] New functionality includes testing.
- [✅ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

#### Test Setup
Created 2 indices - 1 with pluggable dataformat setting and 1 without it

```
PUT /pluggable-test-index
Content-Type: application/json

{
  "settings": {
    "index.pluggable.dataformat.enabled": true,
    "number_of_shards": 1,
    "number_of_replicas": 0
  },
  "mappings": {
    "properties": {
      "timestamp": {"type": "date"},
      "value": {"type": "long"}
    }
  }
}
```
Response (HTTP 200):
```json
{
    "acknowledged": true,
    "shards_acknowledged": true,
    "index": "pluggable-test-index"
}
```

```
PUT /normal-index
Content-Type: application/json

{
  "settings": {
    "number_of_shards": 1,
    "number_of_replicas": 0
  },
  "mappings": {
    "properties": {
      "timestamp": {"type": "date"},
      "value": {"type": "long"}
    }
  }
}
```
Response (HTTP 200):
```json
{
    "acknowledged": true,
    "shards_acknowledged": true,
    "index": "normal-index"
}
```

#### Path 1: Standalone Rollup (`TransportIndexRollupAction`)

**Test 1a: Rollup on pluggable dataformat index — BLOCKED** ✅

Request:
```
PUT /_plugins/_rollup/jobs/rollup-blocked
Content-Type: application/json

{
  "rollup": {
    "enabled": true,
    "schedule": {"interval": {"start_time": 1602100553, "period": 1, "unit": "Minutes"}},
    "last_updated_time": 1602100553,
    "description": "rollup on pluggable index",
    "source_index": "pluggable-test-index",
    "target_index": "rollup-target",
    "page_size": 1,
    "delay": 0,
    "continuous": false,
    "dimensions": [{"date_histogram": {"source_field": "timestamp", "fixed_interval": "1h", "timezone": "UTC"}}],
    "metrics": [{"source_field": "value", "metrics": [{"sum": {}}]}]
  }
}
```

Response (HTTP 400):
```json
{
    "error": {
        "root_cause": [
            {
                "type": "status_exception",
                "reason": "Rollup is not supported on indices with pluggable dataformat enabled"
            }
        ],
        "type": "status_exception",
        "reason": "Rollup is not supported on indices with pluggable dataformat enabled"
    },
    "status": 400
}
```

**Test 1b: Rollup on normal index — ALLOWED** ✅

Request:
```
PUT /_plugins/_rollup/jobs/rollup-normal
Content-Type: application/json

{
  "rollup": {
    "enabled": true,
    "schedule": {"interval": {"start_time": 1602100553, "period": 1, "unit": "Minutes"}},
    "last_updated_time": 1602100553,
    "description": "rollup on normal index",
    "source_index": "normal-index",
    "target_index": "rollup-target-normal",
    "page_size": 1,
    "delay": 0,
    "continuous": false,
    "dimensions": [{"date_histogram": {"source_field": "timestamp", "fixed_interval": "1h", "timezone": "UTC"}}],
    "metrics": [{"source_field": "value", "metrics": [{"sum": {}}]}]
  }
}
```

Response (HTTP 201):
```json
{
    "_id": "rollup-normal",
    "_version": 1,
    "_seq_no": 1,
    "_primary_term": 2,
    "rollup": {
        "rollup_id": "rollup-normal",
        "enabled": true,
        "schedule": {
            "interval": {
                "start_time": 1602100553,
                "period": 1,
                "unit": "Minutes",
                "schedule_delay": 0
            }
        },
        "last_updated_time": 1778595133684,
        "enabled_time": 1778595133684,
        "description": "rollup on normal index",
        "schema_version": 27,
        "source_index": "normal-index",
        "target_index": "rollup-target-normal",
        "metadata_id": null,
        "page_size": 1,
        "delay": 0,
        "continuous": false,
        "dimensions": [
            {
                "date_histogram": {
                    "fixed_interval": "1h",
                    "source_field": "timestamp",
                    "target_field": "timestamp",
                    "timezone": "UTC",
                    "format": null
                }
            }
        ],
        "metrics": [
            {
                "source_field": "value",
                "metrics": [{"sum": {}}]
            }
        ]
    }
}
```

#### Path 2: Standalone Transform (`TransportIndexTransformAction`)

**Test 2a: Transform on pluggable dataformat index — BLOCKED** ✅

Request:
```
PUT /_plugins/_transform/transform-blocked
Content-Type: application/json

{
  "transform": {
    "enabled": true,
    "schedule": {"interval": {"start_time": 1602100553, "period": 1, "unit": "Minutes"}},
    "description": "transform on pluggable index",
    "source_index": "pluggable-test-index",
    "target_index": "transform-target",
    "page_size": 1,
    "data_selection_query": {"match_all": {}},
    "groups": [{"date_histogram": {"source_field": "timestamp", "fixed_interval": "1h", "timezone": "UTC"}}],
    "aggregations": {"sum_value": {"sum": {"field": "value"}}}
  }
}
```

Response (HTTP 400):
```json
{
    "error": {
        "root_cause": [
            {
                "type": "status_exception",
                "reason": "Transform is not supported on indices with pluggable dataformat enabled"
            }
        ],
        "type": "status_exception",
        "reason": "Transform is not supported on indices with pluggable dataformat enabled"
    },
    "status": 400
}
```

**Test 2b: Transform on normal index — ALLOWED** ✅

Request:
```
PUT /_plugins/_transform/transform-normal
Content-Type: application/json

{
  "transform": {
    "enabled": true,
    "schedule": {"interval": {"start_time": 1602100553, "period": 1, "unit": "Minutes"}},
    "description": "transform on normal index",
    "source_index": "normal-index",
    "target_index": "transform-target-normal",
    "page_size": 1,
    "data_selection_query": {"match_all": {}},
    "groups": [{"date_histogram": {"source_field": "timestamp", "fixed_interval": "1h", "timezone": "UTC"}}],
    "aggregations": {"sum_value": {"sum": {"field": "value"}}}
  }
}
```

Response (HTTP 201):
```json
{
    "_id": "transform-normal",
    "_version": 1,
    "_seq_no": 2,
    "_primary_term": 2,
    "transform": {
        "transform_id": "transform-normal",
        "schema_version": 27,
        "schedule": {
            "interval": {
                "start_time": 1602100553,
                "period": 1,
                "unit": "Minutes"
            }
        },
        "metadata_id": null,
        "updated_at": 1778595146233,
        "enabled": true,
        "enabled_at": 1778595146233,
        "description": "transform on normal index",
        "source_index": "normal-index",
        "data_selection_query": {"match_all": {"boost": 1.0}},
        "target_index": "transform-target-normal",
        "page_size": 1,
        "groups": [
            {
                "date_histogram": {
                    "fixed_interval": "1h",
                    "source_field": "timestamp",
                    "target_field": "timestamp",
                    "timezone": "UTC",
                    "format": null
                }
            }
        ],
        "aggregations": {"sum_value": {"sum": {"field": "value"}}},
        "continuous": false
    }
}
```

#### Path 3: ISM Policy Rollup (`AttemptCreateRollupJobStep`)

**Test 3a: ISM Rollup policy on pluggable dataformat index — BLOCKED** ✅

Create policy:
```
PUT /_plugins/_ism/policies/rollup-policy
Content-Type: application/json

{
  "policy": {
    "description": "rollup policy test",
    "default_state": "rollup_state",
    "states": [
      {
        "name": "rollup_state",
        "actions": [
          {
            "rollup": {
              "ism_rollup": {
                "description": "ism rollup test",
                "target_index": "rollup-ism-target",
                "page_size": 1,
                "dimensions": [{"date_histogram": {"source_field": "timestamp", "fixed_interval": "1h", "timezone": "UTC"}}],
                "metrics": [{"source_field": "value", "metrics": [{"sum": {}}]}]
              }
            }
          }
        ],
        "transitions": []
      }
    ]
  }
}
```

Response (HTTP 201):
```json
{
    "_id": "rollup-policy",
    "_version": 1,
    "_primary_term": 2,
    "_seq_no": 7,
    "policy": {
        "policy": {
            "policy_id": "rollup-policy",
            "description": "rollup policy test",
            "default_state": "rollup_state",
            "states": [{"name": "rollup_state", "actions": [{"rollup": {"ism_rollup": {...}}}]}]
        }
    }
}
```

Apply policy:
```
POST /_plugins/_ism/add/pluggable-test-index
Content-Type: application/json

{"policy_id": "rollup-policy"}
```

Response (HTTP 200):
```json
{
    "updated_indices": 1,
    "failures": false,
    "failed_indices": []
}
```

ISM Explain (after ISM executes the step):
```
GET /_plugins/_ism/explain/pluggable-test-index
```

Response:
```json
{
    "pluggable-test-index": {
        "policy_id": "rollup-policy",
        "state": {"name": "rollup_state"},
        "action": {
            "name": "rollup",
            "failed": false,
            "consumed_retries": 1
        },
        "step": {
            "name": "attempt_create_rollup",
            "step_status": "failed"
        },
        "info": {
            "cause": "Rollup is not supported on indices with pluggable dataformat enabled",
            "message": "Failed to validate resolved indices for rollup job"
        },
        "enabled": true
    }
}
```

Server Log:
```
[INFO ][o.o.i.i.ManagedIndexRunner] Executing attempt_create_rollup for pluggable-test-index
[ERROR][o.o.i.i.s.r.AttemptCreateRollupJobStep] Failed to validate resolved indices for rollup job
java.lang.IllegalArgumentException: Rollup is not supported on indices with pluggable dataformat enabled
    at o.o.i.i.step.rollup.AttemptCreateRollupJobStep.validateResolvedIndices(AttemptCreateRollupJobStep.kt:137)
    at o.o.i.i.step.rollup.AttemptCreateRollupJobStep.execute(AttemptCreateRollupJobStep.kt:72)
    at o.o.i.i.ManagedIndexRunner$runManagedIndexConfig$2.invokeSuspend(ManagedIndexRunner.kt:458)
[INFO ][o.o.i.i.ManagedIndexRunner] Finished executing attempt_create_rollup for pluggable-test-index
```

#### Path 4: ISM Policy Transform (`AttemptCreateTransformJobStep`)

**Test 4a: ISM Transform policy on pluggable dataformat index — BLOCKED** ✅

Create policy:
```
PUT /_plugins/_ism/policies/transform-policy
Content-Type: application/json

{
  "policy": {
    "description": "transform policy test",
    "default_state": "transform_state",
    "states": [
      {
        "name": "transform_state",
        "actions": [
          {
            "transform": {
              "ism_transform": {
                "description": "ism transform test",
                "target_index": "transform-ism-target",
                "page_size": 1,
                "data_selection_query": {"match_all": {}},
                "groups": [{"date_histogram": {"source_field": "timestamp", "fixed_interval": "1h", "timezone": "UTC"}}],
                "aggregations": {"sum_value": {"sum": {"field": "value"}}}
              }
            }
          }
        ],
        "transitions": []
      }
    ]
  }
}
```

Response (HTTP 201):
```json
{
    "_id": "transform-policy",
    "_version": 1,
    "_primary_term": 2,
    "_seq_no": 32,
    "policy": {
        "policy": {
            "policy_id": "transform-policy",
            "description": "transform policy test",
            "default_state": "transform_state",
            "states": [{"name": "transform_state", "actions": [{"transform": {"ism_transform": {...}}}]}]
        }
    }
}
```

Apply policy:
```
POST /_plugins/_ism/add/pluggable-test-index
Content-Type: application/json

{"policy_id": "transform-policy"}
```

Response (HTTP 200):
```json
{
    "updated_indices": 1,
    "failures": false,
    "failed_indices": []
}
```

Server Log (after ISM executes the step):
```
[INFO ][o.o.i.i.ManagedIndexRunner] Executing attempt_create_transform for pluggable-test-index
java.lang.IllegalArgumentException: Transform is not supported on indices with pluggable dataformat enabled
```

####  Summary

| # | Path | Failure Case (Blocked) | Success Case (Allowed) |
|---|------|:---:|:---:|
| 1 | Standalone Rollup | ✅ HTTP 400 | ✅ HTTP 201 |
| 2 | Standalone Transform | ✅ HTTP 400 | ✅ HTTP 201 |
| 3 | ISM Policy Rollup | ✅ Step failed with clear error in explain & logs | ✅ Works on normal indices |
| 4 | ISM Policy Transform | ✅ Step failed with clear error in logs | ✅ Works on normal indices |

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/rollup/AttemptCreateRollupJobStep.kt [hunk 1]
```diff
@@ -70,6 +71,16 @@ class AttemptCreateRollupJobStep(private val action: RollupAction) : Step(name)
             // This catches configuration errors early before attempting to create the rollup job.
             validateResolvedIndices(resolvedSourceIndex, resolvedTargetIndex)
 
+            if (PluggableDataFormatUtils.hasPluggableDataFormatEnabled(
```
Comment (tandonks) [inline src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/rollup/AttemptCreateRollupJobStep.kt]: We can also add similar check in AttemptCreateTransformJobStep to fail fast in Transform flow too

Comment (KhishorekumarBS) [reply src/main/kotlin/org/opensearch/indexmanagement/indexstatemanagement/step/rollup/AttemptCreateRollupJobStep.kt]: Added

**Review by KhishorekumarBS — COMMENTED** [review]
Addressed comment

**Review by tandonks — DISMISSED** [review]

**Comment by tandonks** [comment]
LGTM

**Review by tandonks — DISMISSED** [review]

**Review by tandonks — APPROVED** [review]
